code/game/objects/items/RCD.dm

```diff
@@ -572,9 +614,22 @@ RLD
 
 /obj/item/construction/rcd/afterattack(atom/A, mob/user, proximity)
 	. = ..()
-	if(!prox_check(proximity))
-		return
-	rcd_create(A, user)
+	if (mode == RCD_CONVEYOR)	
+		if(!range_check(A, user) || !target_check(A,user)  || istype(A, /obj/machinery/conveyor) || !isopenturf(A) || istype(A, /area/shuttle))
+			to_chat(user, "<span class='warning'>Error! Invalid tile!</span>")
+			return
+		if (!id)
+			to_chat(user, "<span class='warning'>Error! [src] is not linked!</span>")
```

Comment:
Not useful.

Reply:
Conveyor machines should under no condition be generated without a conveyor switch ID assigned to them.

The conveyor switch ID cannot be changed once the machine is placed down, and without a switch ID it will not function.